sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonUtils.scala

```diff
@@ -29,4 +31,28 @@ object JacksonUtils {
       case x => x != stopOn
     }
   }
+
+  /**
+   * Verify if the schema is supported in JSON parsing.
+   */
+  def verifySchema(schema: StructType): Unit = {
+    def verifyType(dataType: DataType): Unit = dataType match {
+      case NullType | BooleanType | ByteType | ShortType | IntegerType | LongType | FloatType |
+           DoubleType | StringType | TimestampType | DateType | BinaryType | _: DecimalType =>
+
+      case st: StructType => st.map(_.dataType).foreach(verifyType)
+
+      case at: ArrayType => verifyType(at.elementType)
+
+      case mt: MapType => verifyType(mt.keyType)
+
+      case udt: UserDefinedType[_] => verifyType(udt.sqlType)
+
+      case _ =>
+        throw new UnsupportedOperationException(
+          s"JSON conversion does not support to process ${dataType.simpleString} type.")
```

Comment:
`does not support to process` is a little hard to parse.  Maybe `Unable to convert column ${name} of type ${dataType.simpleString} to JSON.`